src/Web/DOM/ShadowRoot.purs

```diff
@@ -0,0 +1,47 @@
+module Web.DOM.ShadowRoot
+  ( ShadowRoot
+  , ShadowRootMode (..)
+  , attachShadow
+  , host
+  , mode
+  , toNode
+  ) where
+
+import Prelude
+import Data.Maybe (Maybe(..))
+import Effect (Effect)
+import Unsafe.Coerce (unsafeCoerce)
+import Web.DOM.Internal.Types (Element, Node)
+
+foreign import data ShadowRoot :: Type
+
+data ShadowRootMode = Open | Closed
+
+instance showShadowRootMode :: Show ShadowRootMode where
+  show Open = "open"
+  show Closed = "closed"
+
+attachShadow :: ShadowRootMode -> Element -> Effect ShadowRoot
+attachShadow = _attachShadow <<< modeToProps
```

Comment:
I'd suggest this belongs in `Web.DOM.Element`, since [the member is part of the `Element` interface](https://dom.spec.whatwg.org/#interface-element).

Reply:
Hi Gary, thank you for the suggestion. I have moved the function and related stuff to `Web.DOM.Element` and updated it to accept an argument of type [ShadowRootInit type](https://dom.spec.whatwg.org/#dictdef-shadowrootinit) as per the specs.